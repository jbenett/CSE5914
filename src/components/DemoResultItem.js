import React, { Component } from "react";
import styled from "styled-components";

const Suggestion = styled.b`
  color: ${props => props.theme.primary};
`;

class DemoResultItem extends Component {
  render() {
    return (
      <div className="result-item">
        <div>
          <div className="result-options" onClick={()=>{this.props.removeSuggestion(this.props.id)} }>â“§</div>
          <p>
            "{this.props.contextPrefix}{" "}
            <span className="strike-through">{this.props.original}</span>{" "}
            {this.props.contextSuffix}"
          </p>
          <p>
            "{this.props.contextPrefix}{" "}
            <Suggestion> {this.props.quote} </Suggestion>{" "}
            {this.props.contextSuffix}"
          </p>
          <div className="result-profile">
            <img
              className="profile-items profile-items--image"
              src={this.props.image}
            />
            <p className="profile-items profile-items--text">
              {this.props.author}
            </p>
          </div>
        </div>
      </div>
    );
  }
}

export default DemoResultItem;
