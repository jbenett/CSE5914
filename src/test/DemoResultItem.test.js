import React from 'react';
import { shallow } from 'enzyme';
import DemoResultItem from '../components/DemoResultItem.js'

const myMock = jest.fn();
const clickMock =jest.fn();

describe('Demo Result Item', () => {
  it('should render correctly', () => {
    const component = shallow(<DemoResultItem
      author="author"
      quote="text"
      contextPrefix=""
      original="source"
      image=""
      key="key"
      id="id"
      removeSuggestion={myMock}
      takeSuggestion={myMock}
      onHoverSuggestion={myMock}
    />);
  expect(component).toMatchSnapshot();
  });
  it('should close when the button is clicked', () => {
    const component = shallow(<DemoResultItem
      author="author"
      quote="text"
      contextPrefix=""
      original="source"
      image=""
      key="key"
      id="id"
      removeSuggestion={clickMock}
      takeSuggestion={myMock}
      onHoverSuggestion={myMock}
    />);
    component.find('.result-options').simulate('click');
    expect(clickMock).toHaveBeenCalled();
  })
});
