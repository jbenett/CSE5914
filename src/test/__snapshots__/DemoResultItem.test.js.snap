// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Result Item should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoResultItem
    author="author"
    contextPrefix=""
    id="id"
    image=""
    onHoverSuggestion={[MockFunction]}
    original="source"
    quote="text"
    removeSuggestion={[MockFunction]}
    takeSuggestion={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
          className="result-options"
          onClick={[Function]}
        >
          ⓧ
        </div>
        <p>
          "
          
           
          <span
            className="strike-through"
          >
            source
          </span>
           
          "
        </p>
        <p>
          "
          
           
          <ForwardRef>
             
            text
             
          </ForwardRef>
           
          "
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Take Suggestion
        </button>
        <div
          className="result-profile"
        >
          <img
            className="profile-items profile-items--image"
            src=""
          />
          <p
            className="profile-items profile-items--text"
          >
            author
          </p>
        </div>
      </div>,
      "className": "result-item",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="result-options"
            onClick={[Function]}
          >
            ⓧ
          </div>,
          <p>
            "
            
             
            <span
              className="strike-through"
            >
              source
            </span>
             
            "
          </p>,
          <p>
            "
            
             
            <ForwardRef>
               
              text
               
            </ForwardRef>
             
            "
          </p>,
          <button
            onClick={[Function]}
            type="button"
          >
            Take Suggestion
          </button>,
          <div
            className="result-profile"
          >
            <img
              className="profile-items profile-items--image"
              src=""
            />
            <p
              className="profile-items profile-items--text"
            >
              author
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ⓧ",
            "className": "result-options",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "ⓧ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"",
              "",
              " ",
              <span
                className="strike-through"
              >
                source
              </span>,
              " ",
              undefined,
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "\\"",
            "",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "source",
                "className": "strike-through",
              },
              "ref": null,
              "rendered": "source",
              "type": "span",
            },
            " ",
            undefined,
            "\\"",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"",
              "",
              " ",
              <ForwardRef>
                 
                text
                 
              </ForwardRef>,
              " ",
              undefined,
              "\\"",
            ],
          },
          "ref": null,
          "rendered": Array [
            "\\"",
            "",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  "text",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "text",
                " ",
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "rules": Array [
                    "
  color: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.b",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "b",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            " ",
            undefined,
            "\\"",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Take Suggestion",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Take Suggestion",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="profile-items profile-items--image"
                src=""
              />,
              <p
                className="profile-items profile-items--text"
              >
                author
              </p>,
            ],
            "className": "result-profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "profile-items profile-items--image",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "author",
                "className": "profile-items profile-items--text",
              },
              "ref": null,
              "rendered": "author",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
            className="result-options"
            onClick={[Function]}
          >
            ⓧ
          </div>
          <p>
            "
            
             
            <span
              className="strike-through"
            >
              source
            </span>
             
            "
          </p>
          <p>
            "
            
             
            <ForwardRef>
               
              text
               
            </ForwardRef>
             
            "
          </p>
          <button
            onClick={[Function]}
            type="button"
          >
            Take Suggestion
          </button>
          <div
            className="result-profile"
          >
            <img
              className="profile-items profile-items--image"
              src=""
            />
            <p
              className="profile-items profile-items--text"
            >
              author
            </p>
          </div>
        </div>,
        "className": "result-item",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="result-options"
              onClick={[Function]}
            >
              ⓧ
            </div>,
            <p>
              "
              
               
              <span
                className="strike-through"
              >
                source
              </span>
               
              "
            </p>,
            <p>
              "
              
               
              <ForwardRef>
                 
                text
                 
              </ForwardRef>
               
              "
            </p>,
            <button
              onClick={[Function]}
              type="button"
            >
              Take Suggestion
            </button>,
            <div
              className="result-profile"
            >
              <img
                className="profile-items profile-items--image"
                src=""
              />
              <p
                className="profile-items profile-items--text"
              >
                author
              </p>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ⓧ",
              "className": "result-options",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ⓧ",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "\\"",
                "",
                " ",
                <span
                  className="strike-through"
                >
                  source
                </span>,
                " ",
                undefined,
                "\\"",
              ],
            },
            "ref": null,
            "rendered": Array [
              "\\"",
              "",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "source",
                  "className": "strike-through",
                },
                "ref": null,
                "rendered": "source",
                "type": "span",
              },
              " ",
              undefined,
              "\\"",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "\\"",
                "",
                " ",
                <ForwardRef>
                   
                  text
                   
                </ForwardRef>,
                " ",
                undefined,
                "\\"",
              ],
            },
            "ref": null,
            "rendered": Array [
              "\\"",
              "",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    " ",
                    "text",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "text",
                  " ",
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "rules": Array [
                      "
  color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.b",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "b",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              " ",
              undefined,
              "\\"",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Take Suggestion",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Take Suggestion",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="profile-items profile-items--image"
                  src=""
                />,
                <p
                  className="profile-items profile-items--text"
                >
                  author
                </p>,
              ],
              "className": "result-profile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "profile-items profile-items--image",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "author",
                  "className": "profile-items profile-items--text",
                },
                "ref": null,
                "rendered": "author",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
